modules/Auth/Auth.cpp

```diff
@@ -70,11 +76,35 @@ void Auth::ready() {
         [this](const int32_t evse_index, const StopTransactionRequest& request) {
             this->r_evse_manager.at(evse_index)->call_stop_transaction(request);
         });
-    this->auth_handler->register_reserved_callback([this](const int32_t evse_index, const int32_t& reservation_id) {
-        this->r_evse_manager.at(evse_index)->call_reserve(reservation_id);
-    });
+    this->auth_handler->register_reserved_callback(
+        [this](const std::optional<int32_t> evse_id, const int32_t& reservation_id) {
+            // Only call the evse manager to store the reservation if it is done for a specific evse.
+            if (evse_id.has_value()) {
+                EVLOG_info << "Call reserved callback for evse id " << evse_id.value();
+
+                this->r_evse_manager.at(evse_id.value() - 1)->call_reserve(reservation_id);
```

Comment:
`call_reserve` returns a bool, we should check for the result. Im not sure, but maybe this also allows to move a little bit of the business logic to check whether a reservation can be placed (e.g. in case of faulted, inoperative) to the EvseManager. If I look at `handle_reserve` / `reserve` a few checks are done there

Reply:
Hmm yes that's a good one. Since the ReservationHandler has a copy of this states, it should just always be possible. But ok, if the evse manager returns false here, should we cancel the reservation in that case then? Because I think at this point the reservation has already been made.

Reply:
I just left a todo there for now.

Reply:
The EvseManager shall stay autonoumus with the decision if a reservation can be made, so we should be able to handle the response at some point

Reply:
So should the flow be:
- check in the reservation handler if the reservation can be made, but do not make it yet
- if it can be made, call 'reserve' on the evse manager
- if evse manager returns 'true', we make the reservation in the reservation handler

?

Is that all going well when a new car just arrives in the meantime?